### PR TITLE
Build client in docker

### PR DESCRIPTION
**Pull request by ddimaria**

### CODE REVIEW

File: quadratic-client/package.json [hunk 1]
```diff
@@ -145,6 +147,7 @@
     "sass": "^1.69.5",
     "serve": "^14.2.0",
     "tailwindcss": "^3.3.5",
+    "typescript": "^5.4.5",
```
Comment (jimniels) [inline quadratic-client/package.json:150]: Why was this added? `typescript` is in the root package.json, so does it need to be here?

Comment (ddimaria) [reply quadratic-client/package.json:150]: I was having build issues without it, it might be the same issue we're getting in the next-connections PR

**Review by davidkircos — APPROVED** [review]

File: client.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+COPY ./quadratic-rust-client/. ./quadratic-rust-client/
+COPY ./quadratic-shared/. ./quadratic-shared/
+
+# Run the packaging script for quadratic_py
+RUN ./quadratic-kernels/python-wasm/package.sh --no-poetry
+
+# Build wasm
+WORKDIR /app/quadratic-core
+RUN rustup target add wasm32-unknown-unknown
+RUN echo 'Building wasm...' &&  wasm-pack build --target web --out-dir ../quadratic-client/src/app/quadratic-core --weak-refs
```
Comment (davidkircos) [inline client.Dockerfile:35]: This can also be done form root with `npm run build:wasm:javascript`

File: client.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# Export TS/Rust types
+RUN echo 'Exporting TS/Rust types...' && cargo run --bin export_types
```
Comment (davidkircos) [inline client.Dockerfile:38]: `num run build:wasm:types`

File: client.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+# Build the quadratic-rust-client
+WORKDIR /app
+ARG GIT_COMMIT
+ENV GIT_COMMIT=$GIT_COMMIT
+RUN echo 'Building quadratic-rust-client...' && npm run build --workspace=quadratic-rust-client
```
Comment (davidkircos) [inline client.Dockerfile:44]: `npm run build:rust-client`

**Review by davidkircos — CHANGES_REQUESTED** [review]

File: client.Dockerfile (hunk 2 above)
Comment (ddimaria) [reply client.Dockerfile:35]: That script changes directories, so can't be used.

File: client.Dockerfile (hunk 3 above)
Comment (ddimaria) [reply client.Dockerfile:38]: That script changes directories, so can't be used.

File: client.Dockerfile (hunk 4 above)
Comment (ddimaria) [reply client.Dockerfile:44]: That script changes directories, so can't be used.

**Review by davidkircos — APPROVED** [review]

**Pull request merged**